modin/pandas/series.py

```diff
@@ -1311,9 +1311,51 @@ def reorder_levels(self, order):
         return super(Series, self).reorder_levels(order)
 
     def searchsorted(self, value, side="left", sorter=None):
-        return self._default_to_pandas(
-            pandas.Series.searchsorted, value, side=side, sorter=sorter
-        )
+        """
+        Find indices where elements should be inserted to maintain order.
+
+        Find the indices into a sorted Series self such that, if the
+        corresponding elements in value were inserted before the indices,
+        the order of self would be preserved.
+
+        Parameters
+        ----------
+        value: array_like
+            Values to insert into self.
+        side: {"left", "right"}, optional
+            If "left", the index of the first suitable location found is
+            given. If "right", return the last such index. If there is no
+            suitable index, return either 0 or N (where N is the length of self).
+        sorter: 1-D array_like, optional
+            Optional array of integer indices that sort self into ascending order.
+            They are typically the result of np.argsort.
+
+        Returns
+        -------
+        int or array of int
+            A scalar or array of insertion points with the same shape as value.
+        """
+        # searchsorted should return item number irrespective of Series index, so
+        # Series.index is always set to pandas.RangeIndex, which can be easily processed
+        # on the query_compiler level
+        if not isinstance(self.index, pandas.RangeIndex):
+            self.index = pandas.RangeIndex(start=0, stop=len(self.index))
+        if sorter is not None:
+            self = self.iloc[sorter].reset_index(drop=True)
+            sorter = None
```

Comment:
again about changing source object in place

Reply:
It was found that Modin `iloc` function is not fast enough in comparison with pure Pandas (#1903), so `_default_to_pandas` function is used for now.